nucleus/common/common-util/src/main/java/com/sun/enterprise/loader/ASURLClassLoader.java

```diff
@@ -571,7 +574,9 @@ private void checkManifest(JarFile jar, File file) throws IOException {
 
                 // add to class path of this class loader
                 try {
-                    appendURL(newFile);
+                    if (newFile.exists()) {
```

Comment:
Funny. Here you are adding the existance check which you have removed below.

Reply:
This prevents adding jars to the classpath that don't exist, so the above check isn't needed.